pyrobosim/pyrobosim/gui/world_canvas.py

```diff
@@ -308,19 +286,24 @@ def show_planner_and_path(self, robot=None):
         while self.draw_lock:
             time.sleep(0.001)
         self.draw_lock = True
-        for artist in self.path_planner_artists:
-            artist.remove()
 
         color = robot.color if robot is not None else "m"
-
         if robot and robot.path_planner:
-            self.path_planner_artists = robot.path_planner.plot(
-                self.axes, path_color=color
-            )
-        elif self.world.path_planner:
-            self.path_planner_artists = self.world.path_planner.plot(
-                self.axes, path_color=color
-            )
+            path_planner_artists = robot.path_planner.plot(self.axes, path_color=color)
+            if "graph" in path_planner_artists:
+                for artist in self.path_planner_artists["graph"]:
```

Comment:
A "hack" here is:

```suggestion
            for artist in self.path_planner_artists.get("graph", []):
```

Same for the path artists that follow

Reply:
Oh this is an interesting pattern !



```python
self.path_planner_artists["graph"] = path_planner_artists["graph"]
```
but this line also depends on the check , so even if we use the pattern you suggested we will still need to have a key check.

Reply:
Was able to make use of this pattern somewhere else in the planners to solve a different problem.
Thanks !